### DISCUSSION TITLE
Cannot discover each other in double docker on different host

### DISCUSSION THREAD

**Discussion by ruoruoniao** [eProsima/Fast-DDS#3731] [Support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Communicate as usual.

### Current behavior

Two program cannot discover others, also cannot receive any data.

### Steps to reproduce

Two program (ddshelloworld) is running in two dockers, and each docker is in a independent virtual machine.
Current structure is about this.

![image](https://github.com/eProsima/Fast-DDS/assets/37649564/137edd15-e767-435f-8db2-a93c56eb1366)

Those two dockers can ping to each other by using docker swarm, like this.

![1](https://github.com/eProsima/Fast-DDS/assets/37649564/43144779-8e6c-435b-a8f3-c18fc8d9650f)
![2](https://github.com/eProsima/Fast-DDS/assets/37649564/e5e50e3c-9530-4370-9243-422b2e0e621d)

But those two program cannot discover others, also cannot receive any data.

### Fast DDS version/commit

v2.9.1

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

Default configuration, UDPv4 & SHM

**Comment by ruoruoniao** [comment]
Up to date, I had already been try to remove the network 172.17.x.x, but it also cannot community to others.

**Comment by JLBuenoLopez** [comment]
[DDS Simple Discovery](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/discovery.html#discovery-phases) leverages multicast as per the DDS specification. Have you checked that multicast is enabled with docker swarm mode? If your network architecture is known beforehand you might consider using [Initial Peers](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/simple.html#initial-peers) instead of relying on multicast.

**Comment by ruoruoniao** [comment]
My swarm address is 10.0.1.x, it's flag has a multicast lable. Is that means multicast is enabled?

**Comment by JLBuenoLopez** [comment]
Sorry, I cannot help you with your network configuration. You might consider contacting [Fast DDS commercial support team](https://github.com/eProsima/Fast-DDS#getting-help) to know about the support options available. I am also transferring this ticket to the support discussion forum as per [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md)

**Comment by MagicKim** [comment]
@ruoruoniao Have you solved this problem?
